modules/core/lib/Storage/SQLPermanentStorage.php

```diff
@@ -9,198 +9,210 @@
-		
-		$condition = self::getCondition($type, $key1, $key2);
-		$query = "SELECT * FROM data WHERE " . $condition;
-		$results = $this->db->arrayQuery($query, SQLITE_ASSOC);
-		if (count($results) == 0) return NULL;
-		
-		foreach($results AS $key => $value) {
-			$results[$key]['value'] = unserialize($results[$key]['value']);
-		}
-		return $results;
-	}
-	
-	public function getKeys($type = NULL, $key1 = NULL, $key2 = NULL, $whichKey = 'type') {
-
-		if (!in_array($whichKey, array('key1', 'key2', 'type'), true))
-			throw new Exception('Invalid key type');
-			
-		$condition = self::getCondition($type, $key1, $key2);
-		
-		$query = "SELECT DISTINCT " . $whichKey . " FROM data WHERE " . $condition;
-		$results = $this->db->arrayQuery($query, SQLITE_ASSOC);
-
-		if (count($results) == 0) return NULL;
-		
-		$resarray = array();
-		foreach($results AS $key => $value) {
-			$resarray[] = $value[$whichKey];
-		}
-		
-		return $resarray;
-	}
-	
-	
-	public function remove($type, $key1, $key2) {
-		$query = "DELETE FROM data WHERE " . 
-			"key1 = '" . sqlite_escape_string($key1) . "' AND " . 
-			"key2 = '" . sqlite_escape_string($key2) . "' AND " . 
-			"type = '" . sqlite_escape_string($type) . "'";
-		$results = $this->db->arrayQuery($query, SQLITE_ASSOC);
-		return (count($results) == 1);
-	}
-	
-	public function removeExpired() {
-		$query = "DELETE FROM data WHERE expire NOT NULL AND expire < " . time();
-		$this->db->arrayQuery($query, SQLITE_ASSOC);
-		$changes = $this->db->changes();
-		return $changes;
-	}
-	
-	
-	/**
-	 * Create a SQL condition statement based on parameters
-	 */
-	private static function getCondition($type = NULL, $key1 = NULL, $key2 = NULL) {
-		$conditions = array();
-		
-		if (!is_null($type)) $conditions[] = "type = '" . sqlite_escape_string($type) . "'";
-		if (!is_null($key1)) $conditions[] = "key1 = '" . sqlite_escape_string($key1) . "'";
-		if (!is_null($key2)) $conditions[] = "key2 = '" . sqlite_escape_string($key2) . "'";
-		
-		if (count($conditions) === 0) return '1';
-		
-		$condition = join(' AND ', $conditions);
-		
-		return $condition;
-	}
-	
-	
+class sspmod_core_Storage_SQLPermanentStorage
+{
+    private $db;
+
+    function __construct($name, $config = null)
+    {
+        if (is_null($config)) {
+            $config = SimpleSAML_Configuration::getInstance();
+        }
+
+        $datadir = $config->getPathValue('datadir', 'data/');
+
+        if (!is_dir($datadir)) {
+            throw new Exception('Data directory [' . $datadir. '] does not exist');
+        } else if (!is_writable($datadir)) {
+            throw new Exception('Data directory [' . $datadir. '] is not writable');
+        }
+
+        $sqllitedir = $datadir . 'sqllite/';
+        if (!is_dir($sqllitedir)) {
+            mkdir($sqllitedir);
+        }
+
+        $dbfile = 'sqlite:' . $sqllitedir . $name . '.sqlite';
+        if ($this->db = new \PDO($dbfile)) {
+            $q = @$this->db->query('SELECT key1 FROM data LIMIT 1');
```

Comment:
Error suppression is hardly ever a desirable thing.  I know that the original source code had it as well, but shouldn't failures be loud and obvious?

Reply:
I think I agree with you here, however I'm a bit anxious to break behaviour here.. That would mean we have to wait for the 2.0 release before we can merge this.

Reply:
Now you've made me wonder if I should use the SimpleSAML\Database class here instead of directly calling PDO...

Reply:
At this point I see no easy way to do this, so I'd like to stick with this PR 'as is'.
@thijskh could you please review this?